File: src/main/java/de/janitza/maven/gcs/api/IGoogleCloudStorageService.java
package de.janitza.maven.gcs.api;

import java.io.IOException;
import java.nio.file.Path;
import java.util.Optional;

/**
 * Created with IntelliJ IDEA.
 * User: Jan Müller <jan.mueller@janitza.de>
 * Date: 10.02.15
 * Time: 14:42
 */
public interface IGoogleCloudStorageService {

    void uploadFile(
            Path file,
            Optional<String> relativePathInStorage,
            boolean sharePublic
    ) throws IOException;
}

File: src/main/scala/de/janitza/maven/gcs/api/IGoogleCloudStorageService.scala
package de.janitza.maven.gcs.api

import java.io.IOException
import java.nio.file.Path

/**
  * Created by jan on 11.03.17.
  */
trait IGoogleCloudStorageService {
  def uploadFile(file: Path, relativePathInStorage: Option[String], sharePublic: Boolean): Result[Unit]
}
